<a href="https://colab.research.google.com/github/joajimenez/joajimenez/blob/main/Call_Center_AHT_Prediction_using_Monte_Carlo_simulation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Let's use this notebook to predict a Call Center's Average Handling Time (AHT) by using the Monte Carlo Simulation. The variables are `call_volume`, `agent_skill_level` and `agent_availability`. The AHT can be influenced by the agent skill level and their availability. We want to reduce the AHT.

The variables input to use are `call_volume`: The number of calls received per hour, `agent_skill_level`: The skill level of agents available and `agent_availability`:  The proportion of agents available to take calls.

To simulate AHT we create a simple model that stimates AHT based on agent skill level and availavility. Having a base AHT of `X` minutes, we want to stimate by how much `Y%` and `Z%` it decreases based on `agent_skill_level` and `agent_availability`.

In [16]:
# Let's start by importig Numpy and create our assumptions

import numpy as np

# Example assumptions
base_aht = 8 # base time in minutes to handle the call
skill_level_effect = 0.1 # each skill level unit reduced the AHT by 10%
availability_effect = 0.2 # each 0.1 unit bellow 1 increases AHT by 20%

In [20]:
# Let's generate the random samples

random_seed = np.random.randint(42) # For reproducibility
call_volumes = np.random.randint(300,5000, 1000) # MIN 300, MAX 4999, SIZE 1000
skill_levels = np.random.choice([0,1,2,3],1000) # Random skill levels
availabilities = np.random.uniform(0.1, 1.0, 1000) # Draw random availability by drawing froma uniform distribution

In [21]:
# Simulate AHT for each set of imputs

ahts = []
for cv, sl, av in zip(call_volumes, skill_levels, availabilities):
    aht = base_aht * (1 - sl * skill_level_effect) * (1 + max(0, (1-av)) * availability_effect)
    ahts.append(aht)

In [23]:
# Now we analyse the results
average_aht = np.mean(ahts)
confidence_interval = np.percentile(ahts, [2.5, 97.5])

print(f"Average AHT: {average_aht:.2f} minutes")
print(f"95% Confidence Interval: Calls are predicted to be between {confidence_interval} minutes")

Average AHT: 7.42 minutes
95% Confidence Interval: Calls are predicted to be between [5.71579202 9.31491829] minutes


# Results

The results of the simulation give us two important pieces of information about the average handling time in our call center scenario (based on the input assumptions):

**Average AHT: 7.42 minutes**

This value represents the average handling time across all simulated scenarios. According to the model, it takes around 7.42 minutes to handle a call in our simulated call center. This is a mean derived from the distribution of outcomes across our thousands of simulations, reflecting the combined effects of call volume, agent skill levels, and their availability.

**95% Confidence Interval: Calls are predicted to be between [5.71579202 9.31491829] mins**

**Lower bound (5.71 minutes):**

This is the lower limit of the AHT that we can expect with 95% confidence. It means that, under the assumptions of the model, we can be fairly confident (95% sure) that the average call handling won't be lower than 5.71 mins.

**Upper Bound (9.31 minutes):**

This, on the other hand, is the upper limit of the AHT with the same level of confidence. It indicates that the AHT is unlikely to exceed 9.31 minutes in our simulated scenarios.


##**S.M.A.R.T. Recommendations**

Based on the scenario described and our experience in the call center and education industries, here are some S.M.A.R.T. recommendations that we can apply to reduce the AHT:

**Resource Allocation:**

Specific: Increase the proportion of highly skilled agents available during peak call volume hours by 15% within the next quarter.

Measurable: Track the percentage of highly skilled agents available during peak hours and the average handling time during those periods.

Achievable: Analyze your current staffing levels and skill distribution to determine a feasible increase within budget constraints.

Relevant: This directly addresses the impact of agent skill level on AHT and aligns with your goal of reducing it.

Time-bound: Achieve the 15% increase in highly skilled agent availability within the next three months.

**Performance Monitoring:**

Specific: Implement a system to track individual agent performance metrics, including average handling time and first-call resolution rate, within the next month.

Measurable: Collect data on individual agent performance metrics and analyze trends over time.

Achievable: Utilize existing performance tracking tools or implement a new system with minimal disruption to workflow.

Relevant: Identifying agents with consistently high AHT allows for targeted training or coaching to improve their skills and reduce overall AHT.

Time-bound: Have the performance tracking system operational and collecting data within one month.

**Training and Development:**

Specific: Develop and deliver a training program specifically focused on improving call handling efficiency for all agents within the next six months.

Measurable: Track the completion rate of the training program and assess its impact on individual and team AHT through pre- and post-training comparisons.

Achievable: Partner with your training department or external providers to develop and deliver the program within a realistic timeframe and budget.

Relevant: Equipping agents with the necessary skills and knowledge directly contributes to faster call resolution and reduced AHT.

Time-bound: Complete the training program development and delivery within six months.

**Technology Solutions:**

Specific: Evaluate and implement a call routing system that prioritizes routing calls to highly skilled agents based on real-time availability and skill requirements within the next year.

Measurable: Track the percentage of calls routed to highly skilled agents and compare it to the average handling time for those calls.

Achievable: Conduct a thorough evaluation of available call routing solutions and choose one that fits your budget and technical infrastructure.

Relevant: Automating call routing based on agent skills can significantly improve efficiency and reduce AHT.

Time-bound: Complete the evaluation, selection, and implementation of the call routing system within one year.



## About me

####Hello there.... My name's **Joan Jimenez**.

With 4 years understanding customer needs through call centers and 6 years refining communication skills as an English teacher, I now leverage data analysis and Python to identify and explain complex data, bridging the gap between technical and non-technical audiences. I'm passionate about using data to improve performance and service - let's connect and share insights!

Visit [my LinkedIn profile](https://www.linkedin.com/in/joanjimenezd).